modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx

```diff
@@ -272,22 +310,73 @@ function CardsSection({
 								{'text-secondary': !field}
 							)}
 						>
-							{field?.name ||
-								Liferay.Language.get('not-assigned')}
+							{field
+								? field.label || field.name
+								: Liferay.Language.get('not-assigned')}
 						</p>
 					</ClayInput.GroupItem>
 
 					<ClayInput.GroupItem shrink>
-						<ClayButtonWithIcon
-							aria-label={Liferay.Language.get('assign-field')}
-							displayType="secondary"
-							onClick={onClick}
-							symbol="plus"
-							title={Liferay.Language.get('assign-field')}
+						<UpdateButton
+							field={field}
+							label={label}
+							onClearSelection={onClearSelection}
+							openSelectFieldModal={openSelectFieldModal}
 						/>
 					</ClayInput.GroupItem>
 				</ClayInput.Group>
 			</ClayTable.Cell>
 		</ClayTable.Row>
 	);
 }
+
+interface IUpdateButtonProps {
+	field?: IField;
+	label: string;
+	onClearSelection: () => void;
+	openSelectFieldModal: () => void;
+}
+
+function UpdateButton({
```

Comment:
This is a stateless common component, identical to [the one in List.tsx](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx#L332-L381). As such, we can extract it to a `components` subdir in the right scope, into something like `/js/fds_view/visualization_modes/components/FieldAssignmentControls.tsx`. After this, we can update both List and Cards.

Reply:
Done!